packages/lwc-language-server/src/config.ts

```diff
@@ -74,15 +75,22 @@ export async function onCreatedCustomComponent(context: WorkspaceContext, file:
         try {
             // note, this read/write file must be synchronous, so it is atomic
             const jsconfig: IJsconfig = readJsonSync(jsconfigFile);
+            // deep clone of jsconfig created for update comparison
+            const newJsconfig: IJsconfig = JSON.parse(JSON.stringify(jsconfig));
             if (!jsconfig.compilerOptions) {
-                jsconfig.compilerOptions = {};
+                newJsconfig.compilerOptions = {};
             }
-            jsconfig.compilerOptions.baseUrl = '.';
+
             if (!jsconfig.compilerOptions.paths) {
-                jsconfig.compilerOptions.paths = {};
+                newJsconfig.compilerOptions.paths = {};
+            }
+
+            newJsconfig.compilerOptions.baseUrl = '.';
+            newJsconfig.compilerOptions.paths[moduleTag] = [relativeFilePath];
+
+            if (!equal(jsconfig, newJsconfig)) {
```

Comment:
Do we have perf metrics to tell us if this added any cost? I imagine the files are small enough and this action doesn't happen that often, but curious still

Reply:
The instrumentation for metrics along this code path is not yet implemented, but I agree that this action should be relatively trivial.  It could be an interesting experiment to see the performance between allowing and blocking an identical write.